lib/client.js

```diff
@@ -242,6 +247,47 @@ Client.prototype.method = function(method) {
   });
 };
 
+Client.prototype.stream = function(method) {
+  var args = [].slice.call(arguments, 1);
+  debug('stream', method, args);
+
+  var id = utils.uuid();
+  var stream = new ClientStream(this, id);
+  this._activeStreams[id] = stream;
+
+  this.request('stream', {
+    name: method,
+    args: args,
+    id: id
+  }).catch(err => {
+    this.onstreamerror({ id: id, data: err });
+  });
+
+  return stream;
+};
+
+Client.prototype.onstreamdata = function(broadcast) {
```

Comment:
Might be simpler to remove these three events and replace them with:

```js
Client.prototype.onstreamevent = function(event) {
  this._activeStreams[event.stream].emit(event.type, event.data);
};
```

Then if you need to do any internal house-keeping on any particular events you can bind to their public API just after they're created.

```js
stream
  .on('close', deleteStream);
  .on('error', deleteStream);
```